src/Console/GenerateFactoryCommand.php

```diff
@@ -486,4 +499,43 @@ protected function isLaravel8OrAbove(): bool
 
         return (int) ($version[0] ?? $this->laravel->version())[0] >= 8;
     }
+
+    protected function getFileStructureDiff(string $model): array
+    {
+        return array_diff(explode('\\', $model), explode('/', ucfirst($this->dir)));
```

Comment:
To make it work with custom namespace we need to use it instead of the directory:
```php
if ($this->namespace) {
    return array_diff(explode('\\', $model), explode('\\', $this->namespace));
}

return array_diff(explode('\\', $model), explode('/', ucfirst($this->dir)));
```

Reply:
Of course, that change is only for the file, and something similar need to be implemented for the class to generate namespace prefix - pretty similar to what `generateAdditionalNameSpace` does but with a prefix, not suffix.

Reply:
Thank you for your kind review.

I am not quite sure how everything works with the namespace option.
Could you give me some example about the usage of the namespace option?

Reply:
Of course.
It is quite common to define a different namespace even though you keep the same directory structure.
So in the`composer.json`:
```
"autoload": {
    "psr-4": {
        "CustomNamespace\\": "app/",
        "Tests\\": "tests/",
        "Database\\Factories\\": "database/factories/",
        "Database\\Seeders\\": "database/seeders/"
    },
    "files": [
        "app/MegacamsHelper.php"
    ]
},
```
Then in every class, you would use that namespace instead of `App`. So for models that would be `CustomNamespace\Models`.
Command for factories generation in that case: `php artisan generate:factory --namespace CustomNamespace\\Models --recursive`. It will work but create all factories inside a `CustomNamespace` directory and that is a BC that I mentioned.

Reply:
Thank you for your explanation.

Let me get this straight.
The current problem is that unnecessary directories (not factory) are created when the command: `php artisan generate:factory --namespace CustomNamespace\\Models --recursive` is executed.

Therefore, I need to fix the issue try not to create unnecessary directories inside the namespace that people added as an option.
Am I correct ?

Reply:
@wadakatu, indeed, it is all that is required.

Reply:
@TheDoctor0 

> With the custom namespace, something like CustomNamespace\Models should be treated as a root for models. They should land in database/factories as with default App\Models and from there subdirectories structure for models can be created.

I am a bit confused about the result when a user execute the command: `php artisan generate:factory --namespace CustomNamespace\\Models --recursive`.

What if there is a model directory like below?
What would the result be?

```bash
project
├── README.md
├── app
│   └──Models
│      ├── User.php (namespace: CustomNamespace\Models)
│      ├── UserAuth.php (namespace: CustomNamespace\Models)
│      └── dir
│           ├── FailedJob.php (namespace: CustomNamespace\Models\dir)
│           └── dir_more
│             └── PasswordReset.php (namespace: CustomNamespace\Models\dir\dir_more)
├── database
│   └── factories
└──     
```

In my opinion, the result should be like this.
No need to create `dir` and `dir_more` directories and factories under these directories, because their name space are different than the option: `--namespace CustomNamespace\\Models`.

```bash
project
├── README.md
├── app
│   └──Models
│       ├── User.php (namespace: CustomNamespace\Models)
│       ├── UserAuth.php (namespace: CustomNamespace\Models)
│       └── dir
│           ├── FailedJob.php (namespace: CustomNamespace\Models\dir)
│           └── dir_more
│               └── PasswordReset.php (namespace: CustomNamespace\Models\dir\dir_more)
├── database
│   └── factories
│      ├── UserAuthFactory.php 
│     └── UserFactory.php
└──     
```

Please bear with me for asking so many times.

Reply:
> Please bear with me for asking so many times.

@wadakatu, no need to worry 😄 

```
project
├── README.md
├── app
│   └──Models
│       ├── User.php (namespace: CustomNamespace\Models)
│       ├── UserAuth.php (namespace: CustomNamespace\Models)
│       └── dir
│           ├── FailedJob.php (namespace: CustomNamespace\Models\dir)
│           └── dir_more
│               └── PasswordReset.php (namespace: CustomNamespace\Models\dir\dir_more)
├── database
│   └── factories
│      ├── UserAuthFactory.php 
│     └── UserFactory.php
└──     
```
This is how it currently works without the `--recursive` flag. The namespace that can be provided should be treated as a root namespace for models, so in the recursive mode factories for models in subdirectories definitely should be created.
In theory, you could provide the default `App\Models` as a namespace and you would get a different result than if you skipped and I think that the behavior should be consistent. I hope that you agree with that.

Reply:
@TheDoctor0 
You are right.
I am totally agree with your opinion.

for now, recursive mode behavior is not consistent.
I am going to fix it as soon as possible.

Reply:
@TheDoctor0 

I made some revisions and pushed them.
Could you confirm that it is working as you expected?